### PR TITLE
cmake: Update to Modern CMake usage

### PR DESCRIPTION
**Pull request by noc0lour**
This is still very much a work in progress. But the main parts are working (building libs & python bindings). 

During the update some wrong includes (brackets instead of quotes) were found. 
Static libs & swig building still needs some love.

- [x] convert add_definitions
- [x] split out target_sources()

### CODE REVIEW

File: cmake/Modules/GnuradioConfig.cmake.in [hunk 1]
```diff
@@ -17,13 +17,7 @@
 # the Free Software Foundation, Inc., 51 Franklin Street,
 # Boston, MA 02110-1301, USA.
 
-INCLUDE(FindPkgConfig)
-INCLUDE(FindPackageHandleStandardArgs)
-
-# if GR_REQUIRED_COMPONENTS is not defined, it will be set to the following list (all of them)
-if(NOT GR_REQUIRED_COMPONENTS)
-  set(GR_REQUIRED_COMPONENTS RUNTIME ANALOG BLOCKS DIGITAL FFT FILTER PMT)
-endif()
```
Comment (noc0lour) [inline cmake/Modules/GnuradioConfig.cmake.in]: Removing this was "just" a test, we shouldn't actually remove all of it for compatibility for at least 3.8.

**Comment by noc0lour** [comment]
Well, looks like I need to point my things to the right volk commit (which is not yet in master)'

**Comment by noc0lour** [comment]
OK, now builds and runs all tests. Due to rebasing the volk submodule on master with a hack commit on top it nows fails the `off-by-one` volk tests.

File: gr-analog/lib/CMakeLists.txt [hunk 2]
```diff
@@ -100,8 +78,13 @@ list(APPEND analog_libs
 
 add_library(gnuradio-analog SHARED ${analog_sources})
 target_link_libraries(gnuradio-analog ${analog_libs})
+target_include_directories(gnuradio-analog
+  PUBLIC
+  $<INSTALL_INTERFACE:include>
+  $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/../include>
+  )
 GR_LIBRARY_FOO(gnuradio-analog)
-add_dependencies(gnuradio-analog analog_generated_swigs gnuradio-filter)
```
Comment (noc0lour) [inline gr-analog/lib/CMakeLists.txt]: that should go into a extra commit fixing the usage of non-existent targets (got detected by setting a CMP policy to new.

File: gr-channels/lib/CMakeLists.txt [hunk 3]
```diff
@@ -77,9 +57,13 @@ list(APPEND channels_libs
...
 add_library(gnuradio-channels SHARED ${channels_sources})
 target_link_libraries(gnuradio-channels ${channels_libs})
+target_include_directories(gnuradio-channels
+  PUBLIC
+  $<INSTALL_INTERFACE:include>
+  $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/../include>
+  )
 GR_LIBRARY_FOO(gnuradio-channels)
 add_dependencies(gnuradio-channels
-  channels_generated_swigs
```
Comment (noc0lour) [inline gr-channels/lib/CMakeLists.txt]: extra commit

File: gr-digital/lib/CMakeLists.txt [hunk 4]
```diff
@@ -140,17 +119,20 @@ list(APPEND digital_libs
...
 target_link_libraries(gnuradio-digital ${digital_libs})
+target_include_directories(gnuradio-digital
+  PUBLIC $<INSTALL_INTERFACE:include>
+  PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/../include>
+  )
 GR_LIBRARY_FOO(gnuradio-digital)
 
 add_dependencies(
     gnuradio-digital
-    digital_generated_swigs
```
Comment (noc0lour) [inline gr-digital/lib/CMakeLists.txt]: extra commit

File: gr-filter/lib/CMakeLists.txt [hunk 5]
```diff
@@ -106,16 +84,19 @@ list(APPEND filter_libs
...
 add_library(gnuradio-filter SHARED ${filter_sources})
 target_link_libraries(gnuradio-filter ${filter_libs})
+target_include_directories(gnuradio-filter
+  PUBLIC
+  $<INSTALL_INTERFACE:include>
+  $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/../include>
+  )
 GR_LIBRARY_FOO(gnuradio-filter)
 add_dependencies(gnuradio-filter
-  filter_generated_swigs
```
Comment (noc0lour) [inline gr-filter/lib/CMakeLists.txt]: extra commit

File: gr-trellis/lib/CMakeLists.txt [hunk 6]
```diff
@@ -82,9 +64,13 @@ ENDIF(MSVC)
...
 add_library(gnuradio-trellis SHARED ${trellis_sources})
 target_link_libraries(gnuradio-trellis ${trellis_libs})
+target_include_directories(gnuradio-trellis
+  PUBLIC
+  $<INSTALL_INTERFACE:include>
+  $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/../include>
+  )
 GR_LIBRARY_FOO(gnuradio-trellis)
 add_dependencies(gnuradio-trellis
-  trellis_generated_swigs
```
Comment (noc0lour) [inline gr-trellis/lib/CMakeLists.txt]: extra commit

File: gr-trellis/lib/CMakeLists.txt [hunk 7]
```diff
@@ -102,7 +88,6 @@ if(ENABLE_STATIC_LIBS)
   add_library(gnuradio-trellis_static STATIC ${trellis_sources})
 
   add_dependencies(gnuradio-trellis_static
-    trellis_generated_swigs
```
Comment (noc0lour) [inline gr-trellis/lib/CMakeLists.txt]: extra commit

File: gr-wavelet/lib/CMakeLists.txt [hunk 8]
```diff
@@ -80,7 +76,6 @@ endif()
 
 GR_LIBRARY_FOO(gnuradio-wavelet)
 add_dependencies(gnuradio-wavelet
-  wavelet_generated_swigs
```
Comment (noc0lour) [inline gr-wavelet/lib/CMakeLists.txt]: extra commit

**Comment by noc0lour** [comment]
This is definitely very much work-in-progress, so comments on everything / pointing out bugs and things to do is appreciated. It's basially compiling if the CMake version is new enough.

**Comment by noc0lour** [comment]
Oh, sure, if you can pass me the patch point out commits I can include/factor in, that would be great. The cmake I wrote is not very clean, whitespace-wise.

File: gr-blocks/lib/add_const_bb_impl.cc [hunk 9]
```diff
@@ -24,7 +24,7 @@
 #include "config.h"
 #endif
 
-#include <add_const_bb_impl.h>
+#include "add_const_bb_impl.h"
```
Comment (michaelld) [inline gr-blocks/lib/add_const_bb_impl.cc]: thinking this should be in a different PR

File: gr-blocks/lib/add_const_cc_impl.cc [hunk 10]
```diff
@@ -24,7 +24,7 @@
 #include "config.h"
 #endif
 
-#include <add_const_cc_impl.h>
+#include "add_const_cc_impl.h"
```
Comment (michaelld) [inline gr-blocks/lib/add_const_cc_impl.cc]: thinking this should be in a different PR

File: gr-blocks/lib/add_const_ff_impl.cc [hunk 11]
```diff
@@ -24,7 +24,7 @@
 #include "config.h"
 #endif
 
-#include <add_const_ff_impl.h>
+#include "add_const_ff_impl.h"
```
Comment (michaelld) [inline gr-blocks/lib/add_const_ff_impl.cc]: thinking this should be in a different PR

File: gr-blocks/lib/add_const_ii_impl.cc [hunk 12]
```diff
@@ -24,7 +24,7 @@
 #include "config.h"
 #endif
 
-#include <add_const_ii_impl.h>
+#include "add_const_ii_impl.h"
```
Comment (michaelld) [inline gr-blocks/lib/add_const_ii_impl.cc]: thinking this should be in a different PR

File: gr-blocks/lib/add_const_ss_impl.cc [hunk 13]
```diff
@@ -24,7 +24,7 @@
 #include "config.h"
 #endif
 
-#include <add_const_ss_impl.h>
+#include "add_const_ss_impl.h"
```
Comment (michaelld) [inline gr-blocks/lib/add_const_ss_impl.cc]: thinking this should be in a different PR

File: gr-blocks/lib/add_const_v_impl.cc [hunk 14]
```diff
@@ -24,7 +24,7 @@
 #include "config.h"
 #endif
 
-#include <add_const_v_impl.h>
+#include "add_const_v_impl.h"
```
Comment (michaelld) [inline gr-blocks/lib/add_const_v_impl.cc]: thinking this should be in a different PR

File: gr-blocks/lib/and_const_impl.cc [hunk 15]
```diff
@@ -25,7 +25,7 @@
 #include "config.h"
 #endif
 
-#include <and_const_impl.h>
+#include "and_const_impl.h"
```
Comment (michaelld) [inline gr-blocks/lib/and_const_impl.cc]: thinking this should be in a different PR

File: gr-blocks/lib/float_array_to_int.cc [hunk 16]
```diff
@@ -25,7 +25,7 @@
 #endif
 
 #define _ISOC9X_SOURCE
-#include <float_array_to_int.h>
+#include "float_array_to_int.h"
```
Comment (michaelld) [inline gr-blocks/lib/float_array_to_int.cc]: thinking this should be in a different PR

File: gr-blocks/lib/float_array_to_uchar.cc [hunk 17]
```diff
@@ -25,7 +25,7 @@
 #endif
 
 #define _ISOC9X_SOURCE
-#include <float_array_to_uchar.h>
+#include "float_array_to_uchar.h"
```
Comment (michaelld) [inline gr-blocks/lib/float_array_to_uchar.cc]: thinking this should be in a different PR

File: gr-blocks/lib/integrate_impl.cc [hunk 18]
```diff
@@ -25,7 +25,7 @@
 #include "config.h"
 #endif
 
-#include <integrate_impl.h>
+#include "integrate_impl.h"
```
Comment (michaelld) [inline gr-blocks/lib/integrate_impl.cc]: thinking this should be in a different PR

File: gr-blocks/lib/max_blk_impl.cc [hunk 19]
```diff
@@ -25,7 +25,7 @@
 #include "config.h"
 #endif
 
-#include <max_blk_impl.h>
+#include "max_blk_impl.h"
```
Comment (michaelld) [inline gr-blocks/lib/max_blk_impl.cc]: thinking this should be in a different PR

File: gr-blocks/lib/min_blk_impl.cc [hunk 20]
```diff
@@ -25,7 +25,7 @@
 #include "config.h"
 #endif
 
-#include <min_blk_impl.h>
+#include "min_blk_impl.h"
```
Comment (michaelld) [inline gr-blocks/lib/min_blk_impl.cc]: thinking this should be in a different PR

File: gr-blocks/lib/multiply_by_tag_value_cc_impl.cc [hunk 21]
```diff
@@ -24,7 +24,7 @@
 #include "config.h"
 #endif
 
-#include <multiply_by_tag_value_cc_impl.h>
+#include "multiply_by_tag_value_cc_impl.h"
```
Comment (michaelld) [inline gr-blocks/lib/multiply_by_tag_value_cc_impl.cc]: thinking this should be in a different PR

File: gr-blocks/lib/multiply_conjugate_cc_impl.cc [hunk 22]
```diff
@@ -24,7 +24,7 @@
 #include "config.h"
 #endif
 
-#include <multiply_conjugate_cc_impl.h>
+#include "multiply_conjugate_cc_impl.h"
```
Comment (michaelld) [inline gr-blocks/lib/multiply_conjugate_cc_impl.cc]: thinking this should be in a different PR

File: gr-blocks/lib/multiply_const_v_impl.cc [hunk 23]
```diff
@@ -24,7 +24,7 @@
 #include "config.h"
 #endif
 
-#include <multiply_const_v_impl.h>
+#include "multiply_const_v_impl.h"
```
Comment (michaelld) [inline gr-blocks/lib/multiply_const_v_impl.cc]: thinking this should be in a different PR

File: gr-blocks/lib/or_blk_impl.cc [hunk 24]
```diff
@@ -24,8 +24,8 @@
 #include "config.h"
 #endif
 
+#include "or_blk_impl.h"
```
Comment (michaelld) [inline gr-blocks/lib/or_blk_impl.cc]: thinking this should be in a different PR

File: gr-blocks/lib/probe_signal_v_impl.cc [hunk 25]
```diff
@@ -25,7 +25,7 @@
 #endif
 
 
-#include <probe_signal_v_impl.h>
+#include "probe_signal_v_impl.h"
```
Comment (michaelld) [inline gr-blocks/lib/probe_signal_v_impl.cc]: thinking this should be in a different PR

File: gr-blocks/lib/uchar_array_to_float.cc [hunk 26]
```diff
@@ -20,7 +20,7 @@
  * Boston, MA 02110-1301, USA.
  */
 
-#include <uchar_array_to_float.h>
+#include "uchar_array_to_float.h"
```
Comment (michaelld) [inline gr-blocks/lib/uchar_array_to_float.cc]: thinking this should be in a different PR

File: gr-channels/lib/cfo_model_impl.h [hunk 27]
```diff
@@ -29,7 +29,7 @@
 #include <gnuradio/blocks/multiply.h>
 #include <gnuradio/analog/fastnoise_source.h>
 #include <gnuradio/channels/cfo_model.h>
-#include <sincostable.h>
+#include "sincostable.h"
```
Comment (michaelld) [inline gr-channels/lib/cfo_model_impl.h]: thinking this should be in a different PR

File: gr-channels/lib/fading_model_impl.h [hunk 28]
```diff
@@ -32,7 +32,7 @@
 #include <boost/random.hpp>
 
 #include <gnuradio/fxpt.h>
-#include <sincostable.h>
+#include "sincostable.h"
```
Comment (michaelld) [inline gr-channels/lib/fading_model_impl.h]: thinking this should be in a different PR

File: gr-channels/lib/flat_fader_impl.cc [hunk 29]
```diff
@@ -20,7 +20,7 @@
  * Boston, MA 02110-1301, USA.
  */
 
-#include <flat_fader_impl.h>
+#include "flat_fader_impl.h"
```
Comment (michaelld) [inline gr-channels/lib/flat_fader_impl.cc]: thinking this should be in a different PR

File: gr-channels/lib/flat_fader_impl.cc [hunk 30]
```diff
@@ -83,18 +83,18 @@ namespace gr {
                 float s_q = scale_sin*_GRFASTSIN(d_phi[n]);
                 H += gr_complex(s_i, s_q);
                 }
-    
+
```
Comment (michaelld) [inline gr-channels/lib/flat_fader_impl.cc]: wondering whether this should be in a different PR

File: gr-channels/lib/flat_fader_impl.cc [hunk 31]
```diff
@@ -83,18 +83,18 @@ namespace gr {
...
             if(d_LOS){
                 d_psi[0] = fmod(d_psi[0] + 2*GR_M_PI*d_fDTs*_GRFASTCOS(d_theta_los), 2*GR_M_PI);
                 float los_i = scale_los*_GRFASTCOS(d_psi[0]);
                 float los_q = scale_los*_GRFASTSIN(d_psi[0]);
                 H = H*scale_nlos + gr_complex(los_i,los_q);
                 }
-    
+
```
Comment (michaelld) [inline gr-channels/lib/flat_fader_impl.cc]: wondering whether this should be in a different PR

File: gr-channels/lib/flat_fader_impl.cc [hunk 32]
```diff
@@ -83,18 +83,18 @@ namespace gr {
...
             update_theta();
             Hvec[i] = H;
         }
-        
+
```
Comment (michaelld) [inline gr-channels/lib/flat_fader_impl.cc]: wondering whether this should be in a different PR

File: gr-channels/lib/flat_fader_impl.h [hunk 33]
```diff
@@ -31,7 +31,7 @@
 #include <boost/random.hpp>
 
 #include <gnuradio/fxpt.h>
-#include <sincostable.h>
+#include "sincostable.h"
```
Comment (michaelld) [inline gr-channels/lib/flat_fader_impl.h]: thinking this should be in a different PR

File: gr-channels/lib/selective_fading_model2_impl.cc [hunk 34]
```diff
@@ -21,7 +21,7 @@
  */
 
 #include "selective_fading_model2_impl.h"
-#include <sincostable.h>
+#include "sincostable.h"
```
Comment (michaelld) [inline gr-channels/lib/selective_fading_model2_impl.cc]: thinking this should be in a different PR

File: gr-channels/lib/selective_fading_model2_impl.h [hunk 35]
```diff
@@ -32,7 +32,7 @@
 #include <boost/random.hpp>
 
 #include <gnuradio/fxpt.h>
-#include <sincostable.h>
+#include "sincostable.h"
```
Comment (michaelld) [inline gr-channels/lib/selective_fading_model2_impl.h]: thinking this should be in a different PR

File: gr-channels/lib/selective_fading_model_impl.cc [hunk 36]
```diff
@@ -21,7 +21,7 @@
  */
 
 #include "selective_fading_model_impl.h"
-#include <sincostable.h>
+#include "sincostable.h"
```
Comment (michaelld) [inline gr-channels/lib/selective_fading_model_impl.cc]: thinking this should be in a different PR

File: gr-channels/lib/selective_fading_model_impl.h [hunk 37]
```diff
@@ -32,7 +32,7 @@
 #include <boost/random.hpp>
 
 #include <gnuradio/fxpt.h>
-#include <sincostable.h>
+#include "sincostable.h"
```
Comment (michaelld) [inline gr-channels/lib/selective_fading_model_impl.h]: thinking this should be in a different PR

File: gr-fec/lib/ldpc_bit_flip_decoder_impl.cc [hunk 38]
```diff
@@ -22,7 +22,7 @@
 #include "config.h"
 #endif
 
-#include <ldpc_bit_flip_decoder_impl.h>
+#include "ldpc_bit_flip_decoder_impl.h"
```
Comment (michaelld) [inline gr-fec/lib/ldpc_bit_flip_decoder_impl.cc]: thinking this should be in a different PR

File: gr-fec/lib/ldpc_gen_mtrx_encoder_impl.cc [hunk 39]
```diff
@@ -22,7 +22,7 @@
 #include "config.h"
 #endif
 
-#include <ldpc_gen_mtrx_encoder_impl.h>
+#include "ldpc_gen_mtrx_encoder_impl.h"
```
Comment (michaelld) [inline gr-fec/lib/ldpc_gen_mtrx_encoder_impl.cc]: thinking this should be in a different PR

File: gr-fec/lib/polar_decoder_sc_list.cc [hunk 40]
```diff
@@ -24,10 +24,10 @@
 #include "config.h"
 #endif
 
-#include <gnuradio/io_signature.h>
 #include <gnuradio/fec/polar_decoder_sc_list.h>
+#include "scl_list.h"
```
Comment (michaelld) [inline gr-fec/lib/polar_decoder_sc_list.cc]: thinking this should be in a different PR

File: gr-fec/lib/scl_list.cc [hunk 41]
```diff
@@ -20,7 +20,7 @@
  * Boston, MA 02110-1301, USA.
  */
 
-#include <scl_list.h>
+#include "scl_list.h"
```
Comment (michaelld) [inline gr-fec/lib/scl_list.cc]: thinking this should be in a different PR

File: gr-fec/lib/scl_list.cc [hunk 42]
```diff
@@ -34,7 +34,7 @@ namespace gr {
         scl_list::scl_list(const unsigned int size,
                            const unsigned int block_size,
                            const unsigned int block_power):
-          d_list_size(size), 
+          d_list_size(size),
```
Comment (michaelld) [inline gr-fec/lib/scl_list.cc]: thinking this should be in a different PR

File: cmake/Modules/GnuradioConfig.cmake.in (hunk 1 above)
Comment (noc0lour) [reply cmake/Modules/GnuradioConfig.cmake.in]: We need to discuss/think about how to handle "old" and "modern" CMake. Or if we just require for 3.8 that downstream libraries/packages have to use modern CMake.

File: .gitmodules [hunk 43]
```diff
@@ -1,4 +1,4 @@
 [submodule "volk"]
 	path = volk
-	url = https://github.com/gnuradio/volk.git
+	url = https://github.com/noc0lour/volk.git
```
[the remark this answers is not in the document]
Comment (noc0lour) [reply .gitmodules]: This change is just so `volk` get's it's modern CMake patch. it's just temporarilty. if you actually want to test this with embedded volk you also have to `git submodule sync && git submodule update` a plain update will fail due to the missing commit in the main volk repo.

File: cmake/Modules/FindMPLIB.cmake [hunk 44]
```diff
@@ -25,3 +25,11 @@ set(MPLIB_LIBRARIES ${MPLIBXX_LIBRARY} ${MPLIB_LIBRARY})
...
 find_package_handle_standard_args(MPLIB DEFAULT_MSG MPLIBXX_LIBRARY MPLIB_LIBRARY MPLIB_INCLUDE_DIR)
 mark_as_advanced(MPLIBXX_LIBRARY MPLIB_LIBRARY MPLIB_INCLUDE_DIR)
+
+if (MPLIB_LIBRARIES AND NOT TARGET MPLib::mplib)
+  add_library(MPLib::mplib INTERFACE IMPORTED)
+  set_target_properties(MPLib::mplib PROPERTIES
+    INTERFACE_INCLUDE_DIRECTORIES "${MPLIB_INCLUDE_DIRS}"
+    INTERFACE_LINK_LIBRARIES "${MPLIB_LIBRARIES}"
+  )
+endif()
```
[the remark this answers is not in the document]
Comment (noc0lour) [reply cmake/Modules/FindMPLIB.cmake]: I think we don't have a CMake style GREP or some other guide. Would be nice to nail it down to one definitve right solution.

**Comment by marcusmueller** [comment]
Regarding the VOLK dependency: obviously, we'll have to update VOLK for 3.8, so, can you do a similar PR for your VOLK changes against gnuradio/volk?

**Comment by noc0lour** [comment]
@marcusmueller yes, for volk Maitland already has send me & @n-west a path. I'll post it in a PR.

File: .gitmodules (hunk 43 above)
[the remark this answers is not in the document]
Comment (noc0lour) [reply .gitmodules]: Just updated to a newer git revision of the main volk repository.

**Comment by skoslowski** [comment]
Hi, quickly skimmed over your work here. This is seriously awesome!

In terms of "modern cmake" I noticed you're not using [target_sources()](https://devdocs.io/cmake~3.5/command/target_sources) and [target_add_definitions()](https://devdocs.io/cmake~3.5/command/target_compile_definitions). Any reason for that?

**Comment by noc0lour** [comment]
Ah, I see, instead of generating a list of sources I can just repeatedly call into `target_sources()` and add them to the list.

File: cmake/Modules/GnuradioConfig.cmake.in [hunk 45]
```diff
@@ -17,13 +17,7 @@
...
+include(CMakeFindDependencyMacro)
```
[the remark this answers is not in the document]
Comment (noc0lour) [reply cmake/Modules/GnuradioConfig.cmake.in]: `find_dependency` apparently needs to include to work. For package config cmake files it's advised to use `find_dependency` instead of `find_package` since it's returns on failure and then stops processing.

File: cmake/Modules/FindThrift.cmake [hunk 46]
```diff
@@ -84,3 +84,11 @@ MARK_AS_ADVANCED(
   THRIFT_LIBRARIES THRIFT_INCLUDE_DIRS
   THRIFT_BIN PYTHON_THRIFT_FOUND THRIFT_FOUND
 )
+
+if (THRIFT_FOUND AND NOT TARGET Thrift::thrift)
+  add_library(Thrift::thrift INTERFACE IMPORTED)
+  set_target_properties(Thrift::thrift PROPERTIES
+    INTERFACE_INCLUDE_DIRECTORIES "${THRIFT_INCLUDE_DIRS}"
+    INTERFACE_LINK_LIBRARIES "${THRIFT_LIBRARIES}"
+    )
```
[the remark this answers is not in the document]
Comment (noc0lour) [reply cmake/Modules/FindThrift.cmake]: This has been bugging me as well, my editor is trying to align everything with two spaces.

**Comment by noc0lour** [comment]
Two key questions that came to my mind:
 - What's the use case for building a shared & static library in the same build process ( in fact it's compiling all source files twice anyway).
 - Is there a usecase for using libtool versioning, and when did someone use it the last time to see if it works (whatever that means) ?

**Comment by noc0lour** [comment]
@skoslowski Now all component libraries (if I didn't forget any) are created with their "base" sources and additional sources are added with `target_sources()`. Also defines are now added with `target_compile_definitions` in component CMakeLists. In the toplevel CMakeLists there is still a whole lot of global defines I don't know what to do with. Same for ConfigChecks.cmake
But should be buildable & installable in the current state.

**Comment by skoslowski** [comment]
Awesome, great work

File: cmake/Modules/GnuradioConfig.cmake.in (hunk 1 above)
Comment (noc0lour) [reply cmake/Modules/GnuradioConfig.cmake.in]: From what I've learned/read it's near impossible to support the old variable exporting approach with targets. Basically one would have to render all include paths embedded in exported targets during generation time of the `GnuradioConfig.cmake`. I can't seem to find a way of doing that.

**Comment by jdemel** [comment]
@noc0lour in [this mail](https://www.mail-archive.com/discuss-gnuradio@gnu.org/msg67999.html) you stated that modern cmake is not really a 3.8 feature because it did not get included as part of the feature freeze. Though, #2075 does essentially depend on your work here. As far as I can tell, updates to gr-modtool and by extension to fix #2075, depend on your work in order to make the OOT 3.8 transition as smooth as possible.
This is just my opinion/observation of the topic. Nevertheless, I see a need to clarify on that.

**Comment by noc0lour** [comment]
@jdemel Ultimately it's not my call to make, also I couldn't promise it will be ready "on time". But after confering with @marcusmueller I'll add it to the 3.8 project for inclusion.

**Comment by mbr0wn** [comment]
Do we need CMake 3.8? @marcusmueller and I picked 3.5.1 because it's available in 16.04, which is still relevant.

File: cmake/Modules/GnuradioConfig.cmake.in (hunk 1 above)
Comment (mbr0wn) [reply cmake/Modules/GnuradioConfig.cmake.in]: I would like to stick with the 3.5.1 CMake version we currently have on master.

Comment (mbr0wn) [reply cmake/Modules/GnuradioConfig.cmake.in]: ...unless this totally screws up this PR.

**Comment by mbr0wn** [comment]
Ignore  my comment. I mixed up GR and CMake versions.

**Comment by noc0lour** [comment]
@mbr0wn cmake 3.8 is still needed nonetheless. I had a offline chat with Marcus about that and he was OK with that iirc.

**Comment by drmpeg** [comment]
Since Cmake 3.8 is required, maybe it should be set in the top level CMakeLists.txt? Otherwise, on Ubuntu 16.04 you get many "list does not recognize sub-command FILTER" errors that isn't as useful as just:
```
CMake Error at CMakeLists.txt:30 (cmake_minimum_required):
  CMake 3.8 or higher is required.  You are running version 3.5.1
```

**Comment by noc0lour** [comment]
@mbr0wn there is not a lot to squash here, I think it would be best to use a explanatory merge commit.to have a single commit in the master branch but to preserve the logical structure of the changes.
